src/esp/assets/ResourceManager.cpp

```diff
@@ -2961,7 +2961,30 @@ void ResourceManager::loadTextures(Importer& importer,
         if (image->isCompressed()) {
           format = Mn::GL::textureFormat(image->compressedFormat());
         } else {
-          format = Mn::GL::textureFormat(image->format());
+          const auto pixelFormat = image->format();
+          format = Mn::GL::textureFormat(pixelFormat);
+          // Modify swizzle for single channel textures so that they are
+          // greyscale
+          Mn::UnsignedInt channelCount = pixelFormatChannelCount(pixelFormat);
+          if (channelCount == 1) {
+#ifdef MAGNUM_TARGET_WEBGL
+            ESP_WARNING() << "Single Channel Greyscale Texture : ID" << iTexture
+                          << "incorrectly displays as red instead of "
+                             "greyscale due to greyscale expansion"
+                             "not yet implemented in WebGL.";
+#else
+            currentTexture->setSwizzle<'r', 'r', 'r', '1'>();
```

Comment:
Seems reasonable, but also odd that we need to do this ourselves in ResourceManager. If this is universally correct, it would be nice if Magnum did it automatically @mosra . Related: will we need a similar fixup for the batch renderer?

Reply:
Magnum doesn't do this automatically as it's a choice between:

1. having full control over the input data:
    - either offline preprocessing the JPEG/PNG images to never be single-channel,
    - or for example using the WebM format that is always RGB(A)
    - or by converting the images to a RGB(A) block-compressed format (in other words, Basis-compressed assets won't have this problem)
2. configuring a particular image import plugin to [expand single-channel images to RGB](https://doc.magnum.graphics/magnum/classMagnum_1_1Trade_1_1StbImageImporter.html#Trade-StbImageImporter-configuration)
3. performing a CPU-side expansion on load time (such as broadcasting the red channel to RRR using [Utility::copy()](https://doc.magnum.graphics/corrade/namespaceCorrade_1_1Utility.html#a6cd0c04124a0d7c4879d4cf6da3556a2))
4. using the `ARB_texture_swizzle` extension where supported (as done here)

As a middleware it provides the above four options, but it's on the application developer to pick the one that suits given use case best. On my task list there's some `TextureTools::expandGrayscale()` utility that would make option 3 more convenient, but it's doable with the lower-level APIs as well.

For the batch renderer this fixup is [done in the compositor scripts already](https://github.com/facebookresearch/habitat-sim/blob/07d0b3069068446c886cccb9fae8cffaf0783f2f/src/utils/compositor/CompositorState.cpp#L63). At runtime all images are stored in a (block-compressed) RGBA texture atlas, so this problem isn't there.